### DISCUSSION TITLE
Node affinity rules and pod affinity rules added by defaullt

### DISCUSSION THREAD

**Discussion by rakeshpb90** [strimzi/strimzi-kafka-operator#7159] [Q&A]
We have deployed strimizi operator 0.30.0 and kafka 3.2.0 on the EKS cluster. Node affinity rules and pod affinity seem to be added to the Kafka pod by default. Let me know if this is the default behaviour and if I want to change the behaviour, how can i add that? Let's say if I want to change the topology to "topologyKey: "kubernetes.io/hostname".
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: failure-domain.beta.kubernetes.io/zone
            operator: Exists
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
      - podAffinityTerm:
          labelSelector:
            matchLabels:
              strimzi.io/cluster: test-kafka
              strimzi.io/name: test-kafka-kafka
          topologyKey: failure-domain.beta.kubernetes.io/zone
        weight: 100

**Comment by scholzj** [comment]
Can you share your custom resource? Rules like this are nroamlly added when you configure the rack-awareness and they would you the topology key you configured there.
